Thread stuck in shardingsphere-proxy 5.0.0

The client thread stuck when there have a "重复键违反唯一约束" exception.

version: shardingsphere-proxy 5.0.0

```
import (
	"fmt"
	"gorm.io/driver/postgres"
	"gorm.io/gorm"
	"testing"
)
func TestShardingSphereTransactionRollback(t *testing.T) {
	dsn := "host=127.0.0.1 user=sharding password=sharding dbname=sharding_db port=3307" +
		" sslmode=disable TimeZone=Asia/Shanghai client_encoding=UTF8"
	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
	if err != nil {
		fmt.Println(err)
		return
	}

	db.Debug().Exec("insert into t_order(user_id,order_id,info) values(12,12,'9')")
}
```


![image](https://user-images.githubusercontent.com/13001038/141931696-72edbd57-2ad7-4213-93a8-de2f4ce30434.png)


![image](https://user-images.githubusercontent.com/13001038/141777535-25c19d6b-6225-4beb-9e7e-a9d44104beca.png)

Thanks for using Atomikos! Evaluate http://www.atomikos.com/Main/ExtremeTransactions for advanced features and professional support
or register at http://www.atomikos.com/Main/RegisterYourDownload to disable this message and receive FREE tips & advice
[INFO ] 2021-11-15 19:44:40.995 [main] o.a.s.p.i.i.AbstractBootstrapInitializer - Database name is `PostgreSQL`, version is `9.2.24`
[INFO ] 2021-11-15 19:44:41.227 [main] o.a.s.p.frontend.ShardingSphereProxy - ShardingSphere-Proxy start success.
Thanks for using Atomikos! Evaluate http://www.atomikos.com/Main/ExtremeTransactions for advanced features and professional support
or register at http://www.atomikos.com/Main/RegisterYourDownload to disable this message and receive FREE tips & advice
[INFO ] 2021-11-15 19:48:02.450 [main] o.a.s.p.i.i.AbstractBootstrapInitializer - Database name is `PostgreSQL`, version is `9.2.24`
[INFO ] 2021-11-15 19:48:02.647 [main] o.a.s.p.frontend.ShardingSphereProxy - ShardingSphere-Proxy start success.
[ERROR] 2021-11-15 19:49:19.360 [Connection-2-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
org.postgresql.util.PSQLException: 错误: 重复键违反唯一约束"t_order_0_pkey"
  详细：键值"(order_id)=(12)" 已经存在
    at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
    at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
    at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
    at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
    at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
    at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:307)
    at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:293)
    at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:270)
    at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:266)
    at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:1168)
    at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:103)
    at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
    at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.impl.ProxyStatementExecutorCallback.execute(ProxyStatementExecutorCallback.java:40)
    at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:73)
    at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:66)
    at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:44)
    at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:85)
    at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:64)
    at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.syncExecute(ExecutorEngine.java:101)
    at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.serialExecute(ExecutorEngine.java:87)
    at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.execute(ExecutorEngine.java:81)
    at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:65)
    at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.ProxyJDBCExecutor.execute(ProxyJDBCExecutor.java:66)
    at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.useDriverToExecute(ProxySQLExecutor.java:170)
at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:127)
    at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:116)
    at org.apache.shardingsphere.proxy.backend.communication.ProxyLockEngine.doExecute(ProxyLockEngine.java:103)
    at org.apache.shardingsphere.proxy.backend.communication.ProxyLockEngine.execute(ProxyLockEngine.java:81)
    at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.execute(DatabaseCommunicationEngine.java:99)
    at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:55)
    at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.text.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:63)
    at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:93)
    at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:71)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Exception in thread "Connection-2-ThreadExecutor" java.lang.IllegalArgumentException: No enum constant org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel.错误
    at java.lang.Enum.valueOf(Enum.java:238)
    at org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel.valueOf(PostgreSQLMessageSeverityLevel.java:25)
    at org.apache.shardingsphere.proxy.frontend.postgresql.err.PostgreSQLErrPacketFactory.createErrorResponsePacket(PostgreSQLErrPacketFactory.java:75)
    at org.apache.shardingsphere.proxy.frontend.postgresql.err.PostgreSQLErrPacketFactory.newInstance(PostgreSQLErrPacketFactory.java:48)
    at org.apache.shardingsphere.proxy.frontend.postgresql.command.PostgreSQLCommandExecuteEngine.getErrorPacket(PostgreSQLCommandExecuteEngine.java:72)
    at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.processException(CommandExecutorTask.java:108)
    at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:75)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)

@menghaoranss, my doubt is why client thread stuck when use "tx := db.Begin()". shardingshpere-proxy log is below.

[INFO ] 2021-11-16 15:21:41.528 [Connection-1-ThreadExecutor] ShardingSphere-SQL - Logic SQL: insert into t_order(user_id,order_id,info) values(12,12,'9')
[INFO ] 2021-11-16 15:21:41.528 [Connection-1-ThreadExecutor] ShardingSphere-SQL - SQLStatement: PostgreSQLInsertStatement(withSegment=Optional.empty)
[INFO ] 2021-11-16 15:21:41.528 [Connection-1-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: insert into t_order_0(user_id,order_id,info) values(12, 12, '9')
[ERROR] 2021-11-16 15:21:41.568 [Connection-1-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
org.postgresql.util.PSQLException: 错误: 重复键违反唯一约束"t_order_0_pkey"
  详细：键值"(order_id)=(12)" 已经存在
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:307)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:293)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:270)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:266)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:1168)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:103)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.impl.ProxyStatementExecutorCallback.execute(ProxyStatementExecutorCallback.java:40)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:73)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:66)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:44)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute$original$8MO613Zd(JDBCExecutorCallback.java:85)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute$original$8MO613Zd$accessor$p2QdcgCS(JDBCExecutorCallback.java)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback$auxiliary$RQoBW4Um.call(Unknown Source)
	at org.apache.shardingsphere.agent.core.plugin.interceptor.InstanceMethodAroundInterceptor.intercept(InstanceMethodAroundInterceptor.java:75)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:64)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.syncExecute(ExecutorEngine.java:101)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.serialExecute(ExecutorEngine.java:87)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.execute(ExecutorEngine.java:81)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:65)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.ProxyJDBCExecutor.execute(ProxyJDBCExecutor.java:71)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.useDriverToExecute(ProxySQLExecutor.java:185)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:138)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:127)
	at org.apache.shardingsphere.proxy.backend.communication.ProxyLockEngine.doExecute(ProxyLockEngine.java:103)
	at org.apache.shardingsphere.proxy.backend.communication.ProxyLockEngine.execute(ProxyLockEngine.java:81)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.execute(DatabaseCommunicationEngine.java:126)
	at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:55)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.text.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:71)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:99)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run$original$jfB9aWX5(CommandExecutorTask.java:72)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run$original$jfB9aWX5$accessor$dGYE9mDY(CommandExecutorTask.java)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask$auxiliary$8LCUPbL5.call(Unknown Source)
	at org.apache.shardingsphere.agent.core.plugin.interceptor.InstanceMethodAroundInterceptor.intercept(InstanceMethodAroundInterceptor.java:75)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Exception in thread "Connection-1-ThreadExecutor" java.lang.IllegalArgumentException: No enum constant org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel.错误
	at java.lang.Enum.valueOf(Enum.java:238)
	at org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel.valueOf(PostgreSQLMessageSeverityLevel.java:25)
	at org.apache.shardingsphere.proxy.frontend.postgresql.err.PostgreSQLErrPacketFactory.createErrorResponsePacket(PostgreSQLErrPacketFactory.java:76)
	at org.apache.shardingsphere.proxy.frontend.postgresql.err.PostgreSQLErrPacketFactory.newInstance(PostgreSQLErrPacketFactory.java:48)
	at org.apache.shardingsphere.proxy.frontend.postgresql.command.PostgreSQLCommandExecuteEngine.getErrorPacket(PostgreSQLCommandExecuteEngine.java:82)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.processException$original$jfB9aWX5(CommandExecutorTask.java:117)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.processException$original$jfB9aWX5$accessor$dGYE9mDY(CommandExecutorTask.java)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask$auxiliary$pPB5Sj50.call(Unknown Source)
	at org.apache.shardingsphere.agent.core.plugin.interceptor.InstanceMethodAroundInterceptor.intercept(InstanceMethodAroundInterceptor.java:75)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.processException(CommandExecutorTask.java)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run$original$jfB9aWX5(CommandExecutorTask.java:76)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run$original$jfB9aWX5$accessor$dGYE9mDY(CommandExecutorTask.java)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask$auxiliary$8LCUPbL5.call(Unknown Source)
	at org.apache.shardingsphere.agent.core.plugin.interceptor.InstanceMethodAroundInterceptor.intercept(InstanceMethodAroundInterceptor.java:75)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacket.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacket.java
@@ -69,9 +69,9 @@ public final class OpenGaussErrorResponsePacket implements PostgreSQLIdentifierP
         fillRequiredFieldsIfNecessary();
     }
     
-    public OpenGaussErrorResponsePacket(final PostgreSQLMessageSeverityLevel severityLevel, final String sqlState, final String message) {
+    public OpenGaussErrorResponsePacket(final String severityLevel, final String sqlState, final String message) {
         fields = new LinkedHashMap<>(4, 1);
-        fields.put(FIELD_TYPE_SEVERITY, severityLevel.name());
+        fields.put(FIELD_TYPE_SEVERITY, severityLevel);
         fields.put(FIELD_TYPE_CODE, sqlState);
         fields.put(FIELD_TYPE_MESSAGE, message);
         fillRequiredFieldsIfNecessary();

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacket.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/main/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacket.java
@@ -17,7 +17,6 @@
 
 package org.apache.shardingsphere.db.protocol.opengauss.packet.command.generic;
 
-import org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLIdentifierPacket;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLIdentifierTag;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLMessagePacketType;

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/test/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacketTest.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-opengauss/src/test/java/org/apache/shardingsphere/db/protocol/opengauss/packet/command/generic/OpenGaussErrorResponsePacketTest.java
@@ -83,7 +83,7 @@ public final class OpenGaussErrorResponsePacketTest {
     
     private OpenGaussErrorResponsePacket createErrorResponsePacket() {
         Map<Character, String> serverErrorMessages = new LinkedHashMap<>();
-        serverErrorMessages.put(OpenGaussErrorResponsePacket.FIELD_TYPE_SEVERITY, PostgreSQLMessageSeverityLevel.FATAL.name());
+        serverErrorMessages.put(OpenGaussErrorResponsePacket.FIELD_TYPE_SEVERITY, PostgreSQLMessageSeverityLevel.FATAL);
         serverErrorMessages.put(OpenGaussErrorResponsePacket.FIELD_TYPE_CODE, PostgreSQLErrorCode.INVALID_CATALOG_NAME.getErrorCode());
         serverErrorMessages.put(OpenGaussErrorResponsePacket.FIELD_TYPE_MESSAGE, "database \"test\" does not exist");
         serverErrorMessages.put(OpenGaussErrorResponsePacket.FIELD_TYPE_ERRORCODE, "-1");

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/constant/PostgreSQLMessageSeverityLevel.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/constant/PostgreSQLMessageSeverityLevel.java
@@ -22,7 +22,7 @@ package org.apache.shardingsphere.db.protocol.postgresql.constant;
  *
  * @see <a href="https://www.postgresql.org/docs/12/runtime-config-logging.html#RUNTIME-CONFIG-SEVERITY-LEVELS">Table 19.2. Message Severity Levels</a>
  */
-public class PostgreSQLMessageSeverityLevel {
+public final class PostgreSQLMessageSeverityLevel {
     
     public static final String DEBUG1 = "DEBUG1";
 

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/constant/PostgreSQLMessageSeverityLevel.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/constant/PostgreSQLMessageSeverityLevel.java
@@ -22,29 +22,29 @@ package org.apache.shardingsphere.db.protocol.postgresql.constant;
  *
  * @see <a href="https://www.postgresql.org/docs/12/runtime-config-logging.html#RUNTIME-CONFIG-SEVERITY-LEVELS">Table 19.2. Message Severity Levels</a>
  */
-public enum PostgreSQLMessageSeverityLevel {
+public class PostgreSQLMessageSeverityLevel {
     
-    DEBUG1,
-    
-    DEBUG2,
-    
-    DEBUG3,
-    
-    DEBUG4,
-    
-    DEBUG5,
-    
-    INFO,
-    
-    NOTICE,
-    
-    WARNING,
-    
-    ERROR,
-    
-    LOG,
-    
-    FATAL,
-    
-    PANIC
+    public static final String DEBUG1 = "DEBUG1";
+
+    public static final String DEBUG2 = "DEBUG2";
+
+    public static final String DEBUG3 = "DEBUG3";
+
+    public static final String DEBUG4 = "DEBUG4";
+
+    public static final String DEBUG5 = "DEBUG5";
+
+    public static final String INFO = "INFO";
+
+    public static final String NOTICE = "NOTICE";
+
+    public static final String WARNING = "WARNING";
+
+    public static final String ERROR = "ERROR";
+
+    public static final String LOG = "LOG";
+
+    public static final String FATAL = "FATAL";
+
+    public static final String PANIC = "PANIC";
 }

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java
@@ -112,7 +112,7 @@ public final class PostgreSQLErrorResponsePacket implements PostgreSQLIdentifier
      * @return PostgreSQL error response packet builder
      * @see <a href="https://www.postgresql.org/docs/12/protocol-error-fields.html">52.8. Error and Notice Message Fields</a>
      */
-    public static Builder newBuilder(final PostgreSQLMessageSeverityLevel severity, final PostgreSQLErrorCode postgreSQLErrorCode, final String message) {
+    public static Builder newBuilder(final String severity, final PostgreSQLErrorCode postgreSQLErrorCode, final String message) {
         return newBuilder(severity, postgreSQLErrorCode.getErrorCode(), message);
     }
     
@@ -125,7 +125,7 @@ public final class PostgreSQLErrorResponsePacket implements PostgreSQLIdentifier
      * @return PostgreSQL error response packet builder
      * @see <a href="https://www.postgresql.org/docs/12/protocol-error-fields.html">52.8. Error and Notice Message Fields</a>
      */
-    public static Builder newBuilder(final PostgreSQLMessageSeverityLevel severity, final String code, final String message) {
+    public static Builder newBuilder(final String severity, final String code, final String message) {
         return new Builder(severity, code, message);
     }
     
@@ -133,12 +133,12 @@ public final class PostgreSQLErrorResponsePacket implements PostgreSQLIdentifier
         
         private final Map<Character, String> fields = new LinkedHashMap<>(16, 1);
         
-        private Builder(final PostgreSQLMessageSeverityLevel severity, final String code, final String message) {
+        private Builder(final String severity, final String code, final String message) {
             Preconditions.checkArgument(null != severity, "The severity is always present!");
             Preconditions.checkArgument(!Strings.isNullOrEmpty(code), "The SQLSTATE code is always present!");
             Preconditions.checkArgument(!Strings.isNullOrEmpty(message), "The message is always present!");
-            fields.put(FIELD_TYPE_SEVERITY, severity.name());
-            fields.put(FIELD_TYPE_SEVERITY_NON_LOCALIZED, severity.name());
+            fields.put(FIELD_TYPE_SEVERITY, severity);
+            fields.put(FIELD_TYPE_SEVERITY_NON_LOCALIZED, severity);
             fields.put(FIELD_TYPE_CODE, code);
             fields.put(FIELD_TYPE_MESSAGE, message);
         }

--- a/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java
+++ b/shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java
@@ -20,7 +20,6 @@ package org.apache.shardingsphere.db.protocol.postgresql.packet.generic;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
 import org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLErrorCode;
-import org.apache.shardingsphere.db.protocol.postgresql.constant.PostgreSQLMessageSeverityLevel;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLIdentifierPacket;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLIdentifierTag;
 import org.apache.shardingsphere.db.protocol.postgresql.packet.identifier.PostgreSQLMessagePacketType;

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactory.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactory.java
@@ -73,7 +73,7 @@ public final class PostgreSQLErrPacketFactory {
     }
     
     private static PostgreSQLErrorResponsePacket createErrorResponsePacket(final ServerErrorMessage serverErrorMessage) {
-        return PostgreSQLErrorResponsePacket.newBuilder(PostgreSQLMessageSeverityLevel.valueOf(serverErrorMessage.getSeverity()), serverErrorMessage.getSQLState(), serverErrorMessage.getMessage())
+        return PostgreSQLErrorResponsePacket.newBuilder(serverErrorMessage.getSeverity(), serverErrorMessage.getSQLState(), serverErrorMessage.getMessage())
                 .detail(serverErrorMessage.getDetail()).hint(serverErrorMessage.getHint()).position(serverErrorMessage.getPosition())
                 .internalQueryAndInternalPosition(serverErrorMessage.getInternalQuery(), serverErrorMessage.getInternalPosition()).where(serverErrorMessage.getWhere())
                 .schemaName(serverErrorMessage.getSchema()).tableName(serverErrorMessage.getTable()).columnName(serverErrorMessage.getColumn()).dataTypeName(serverErrorMessage.getDatatype())

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactoryTest.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactoryTest.java
@@ -37,7 +37,7 @@ public final class PostgreSQLErrPacketFactoryTest {
     @Test
     public void assertPSQLExceptionWithServerErrorMessageNotNull() throws NoSuchFieldException, IllegalAccessException {
         ServerErrorMessage serverErrorMessage = mock(ServerErrorMessage.class);
-        when(serverErrorMessage.getSeverity()).thenReturn(PostgreSQLMessageSeverityLevel.FATAL.name());
+        when(serverErrorMessage.getSeverity()).thenReturn(PostgreSQLMessageSeverityLevel.FATAL);
         when(serverErrorMessage.getSQLState()).thenReturn("sqlState");
         when(serverErrorMessage.getMessage()).thenReturn("message");
         when(serverErrorMessage.getPosition()).thenReturn(1);
@@ -45,7 +45,7 @@ public final class PostgreSQLErrPacketFactoryTest {
         Field packetField = PostgreSQLErrorResponsePacket.class.getDeclaredField("fields");
         packetField.setAccessible(true);
         Map<Character, String> fields = (Map<Character, String>) packetField.get(actual);
-        assertThat(fields.get(PostgreSQLErrorResponsePacket.FIELD_TYPE_SEVERITY), is(PostgreSQLMessageSeverityLevel.FATAL.name()));
+        assertThat(fields.get(PostgreSQLErrorResponsePacket.FIELD_TYPE_SEVERITY), is(PostgreSQLMessageSeverityLevel.FATAL));
         assertThat(fields.get(PostgreSQLErrorResponsePacket.FIELD_TYPE_CODE), is("sqlState"));
         assertThat(fields.get(PostgreSQLErrorResponsePacket.FIELD_TYPE_MESSAGE), is("message"));
         assertThat(fields.get(PostgreSQLErrorResponsePacket.FIELD_TYPE_POSITION), is("1"));

